packages/shared/src/server/definitions.ts

```diff
@@ -172,3 +172,153 @@ export const convertScoreReadToInsert = (
     event_ts: new Date(record.event_ts).getTime(),
   };
 };
+
+/**
+ * Expects a single record from a `select * from traces` query. Must be a raw query to keep original
+ * column names, not the Prisma mapped names.
+ */
+export const convertPostgresTraceToInsert = (
+  trace: Record<string, any>,
+): TraceRecordInsertType => {
+  return {
+    id: trace.id,
+    timestamp:
+      trace.timestamp?.toISOString().replace("T", " ").slice(0, -1) ?? null,
+    name: trace.name,
+    user_id: trace.user_id,
+    metadata:
+      typeof trace.metadata === "string"
+        ? { metadata: trace.metadata }
+        : Array.isArray(trace.metadata)
+          ? { metadata: trace.metadata }
+          : trace.metadata,
+    release: trace.release,
+    version: trace.version,
+    project_id: trace.project_id,
+    public: trace.public,
+    bookmarked: trace.bookmarked,
+    tags: trace.tags,
+    input: trace.input,
+    output: trace.output,
+    session_id: trace.session_id,
+    created_at:
+      trace.created_at?.toISOString().replace("T", " ").slice(0, -1) ?? null,
+    updated_at:
+      trace.updated_at?.toISOString().replace("T", " ").slice(0, -1) ?? null,
+    event_ts:
+      trace.timestamp?.toISOString().replace("T", " ").slice(0, -1) ?? null,
```

Comment:
New is that we also set the event_ts in the migrations here (timestamp for trace, score; start_time for observations).
@maxdeichmann We initially said that we'd just default to 1970-01-01 and overwrite - but per our discussion overwriting is not an option and would produce duplicates instead. Hence, I try to infer the domain timestamp right away to set this.